modules/OFConnectionManager2/module/src/ofconnectionmanager.c

```diff
@@ -1118,8 +1119,13 @@ ind_cxn_set_aux_cxns(connection_t *main_cxn, uint32_t num_aux)
             return 1;
         }
     } else {
+        controller_t *controller = main_cxn->controller;
         for (idx = main_cxn->controller->num_aux+1; idx <= num_aux; ++idx) {
             if (controller_add_cxn(main_cxn->controller, idx) == NULL) {
+                while (--idx > controller->num_aux) {
+                    controller_stop_cxn(controller->cxns[idx]);
+                    controller->cxns[idx] = NULL;
+                }
```

Comment:
We could call remove_aux_cxns but we'd have to mess with controller->num_aux.

Calling controller_disconnect seems a little excessive since we can report the failure in the set_aux_cxns_reply message.